assets/src/edit-story/app/prepublish/errors/metadata.js

```diff
@@ -0,0 +1,167 @@
+ * If the story does not have a title, return an error message.
+ * Otherwise, return undefined.
+ *
+ * @param {Story} story The story being checked for critical metadata
+ * @return {Guidance|undefined} Guidance object for consumption
+ */
+export function storyTitle(story) {
+  if (typeof story.title !== 'string' || story.title.trim() === '') {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.ERROR,
+      storyId: story.storyId,
+      message: __('Missing story title', 'web-stories'),
+    };
+  }
+  return undefined;
+}
+
+/**
+ * Check that the story's cover resource has sufficient dimensions.
+ * If the resource is too small in either dimension, return an error message.
+ * Otherwise, return undefined.
+ *
+ * @param {Story} story The story being checked for critical metadata
+ * @return {Guidance|undefined} Guidance object for consumption
+ */
+export function storyCoverPortraitSize(story) {
+  if (
+    story.featuredMedia.height < FEATURED_MEDIA_RESOURCE_MIN_HEIGHT ||
+    story.featuredMedia.width < FEATURED_MEDIA_RESOURCE_MIN_WIDTH
+  ) {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.ERROR,
+      storyId: story.storyId,
+      message: __("Story's portrait cover image is too small", 'web-stories'),
+    };
+  }
+  return undefined;
+}
+
+/**
+ * Check that the story's publisher logo resource has sufficient dimensions.
+ * If the resource is too small in either dimension, return an error message.
+ * Otherwise, return undefined.
+ *
+ * @param {Story} story The story being checked for critical metadata
+ * @return {Guidance|undefined} Guidance object for consumption
+ */
+export function publisherLogoSize(story) {
+  if (
+    story.publisherLogo.height < PUBLISHER_LOGO_MIN_HEIGHT ||
+    story.publisherLogo.width < PUBLISHER_LOGO_MIN_WIDTH
+  ) {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.ERROR,
+      storyId: story.storyId,
+      message: __("Story's publisher logo image is too small", 'web-stories'),
+    };
+  }
+  return undefined;
+}
+
+/**
+ * Check for link and page attachment conflicts.
+ * If there is an element with a link in the page attachment region, return an error message.
+ * Otherwise, return undefined.
+ *
+ * @param {Story} story The story being checked for critical metadata
+ * @return {Guidance|undefined} Guidance object for consumption
+ */
+export function linkInPageAttachmentRegion(story) {
+  const { pages } = story;
+  const isLinkInPageAttachmentArea = pages.some((page) => {
+    const { elements } = page;
+    const isLinkAttached = Boolean(page?.pageAttachment?.url.length);
+    return (
+      isLinkAttached &&
+      elements.filter(({ link }) => link?.url?.length).some(isElementBelowLimit)
+    );
+  });
+
+  if (isLinkInPageAttachmentArea) {
+    return {
+      type: PRE_PUBLISH_MESSAGE_TYPES.ERROR,
+      storyId: story.storyId,
+      message: __(
+        'Story has a link in the page attachment region',
+        'web-stories'
+      ),
+    };
+  }
```

Comment:
Don't we care more about the page ID where this is happening rather than the story ID?

What if this happens on multiple pages?

As a user I would want to quickly find & fix the page with this issue.

Reply:
yes! I'll have to return an array of page IDs